mscore/prefsdialog.cpp

```diff
@@ -938,6 +988,125 @@ void PreferenceDialog::updateBgView(bool useColor)
             }
       }
 
+
+//---------------------------------------------------------
+//   languageUpdate
+//---------------------------------------------------------
+
+void PreferenceDialog::languageUpdate()
+      {
+      language->blockSignals(true);
+      language->clear();
+      QString lang = preferences.getString(PREF_UI_APP_LANGUAGE);
+      cout << lang.toStdString() << endl;
+      int curIdx = 0;
+      for(int i = 0; i < mscore->languages().size(); ++i) {
+            language->addItem(mscore->languages().at(i).name, i);
+            if (mscore->languages().at(i).key == lang)
+                  curIdx = i;
+            }
+      language->blockSignals(false);
+      language->setCurrentIndex(curIdx);
+      }
+
+//---------------------------------------------------------
+//   languageUpdate
+//---------------------------------------------------------
+
+void PreferenceDialog::languageApply()
+      {
+      int lang = language->itemData(language->currentIndex()).toInt();
+      QString l = lang == 0 ? "system" : mscore->languages().at(lang).key;
+      bool languageChanged = l != preferences.getString(PREF_UI_APP_LANGUAGE);
+      preferences.setPreference(PREF_UI_APP_LANGUAGE, l);
```

Comment:
shouldn't `preferences.setPreference(PREF_UI_APP_LANGUAGE, l);` be in the `if (languageChanged)` block?

Reply:
Please review this before changing it. It seemed odd to me, but i haven't analysed it thoroughly. There's probably a reason why it was put that way.